src/utils/i18n.js

```diff
@@ -6,11 +6,37 @@
 
 import annotationsLocaleData from 'box-annotations-locale-data'; // eslint-disable-line
 import boxElementsMessages from 'box-elements-messages'; // eslint-disable-line
-import localeData from 'react-intl-locale-data'; // eslint-disable-line
+import localeData from 'react-intl-locale-data';
+import { IntlProvider, addLocaleData } from 'react-intl';
 
 declare var __LANGUAGE__: string;
 
-const language = __LANGUAGE__;
+let language = __LANGUAGE__;
 const messages = { ...annotationsLocaleData, ...boxElementsMessages };
+const getLocale = (locale?: string) => {
+    let lang = locale;
 
-export default { language, messages, localeData };
+    if (!lang) {
+        lang = language;
+    }
+
+    return lang.substr(0, lang.indexOf('-'));
+};
+
+const createIntlProvider = ({ provider, language: lang, intlLocaleData }: IntlOptions): IntlContext => {
+    addLocaleData(intlLocaleData || localeData);
```

Comment:
Is this || needed? It seems like the data would always be the same, since it's always coming from react-intl.